lib/oli_web/live_session_plugs/set_sidebar.ex

```diff
@@ -25,12 +25,18 @@ defmodule OliWeb.LiveSessionPlugs.SetSidebar do
       socket =
         attach_hook(socket, :sidebar_hook, :handle_params, fn
           params, uri, socket ->
-            sidebar_expanded_from_assigns = socket.assigns.sidebar_expanded
-            sidebar_expanded_from_params = string_to_boolean(params["sidebar_expanded"] || "true")
+            sidebar_from_assigns = socket.assigns.sidebar_expanded
+            sidebar_from_params = string_to_boolean(params["sidebar_expanded"] || "true")
 
-            socket = assign(socket, uri: uri, sidebar_expanded: sidebar_expanded_from_params)
+            socket = assign(socket, uri: uri, sidebar_expanded: sidebar_from_params)
 
-            if sidebar_expanded_from_assigns != sidebar_expanded_from_params do
+            previous_lv_url = socket.private[:connect_params]["_live_referer"]
+            current_lv_url = uri
+
+            has_sidebar_changed = sidebar_from_assigns != sidebar_from_params
+            is_same_workspace = is_same_workspace(previous_lv_url, current_lv_url)
+
+            if is_same_workspace and has_sidebar_changed do
```

Comment:
I'll pass on this because it clearly expresses my intention to check that we are in the same workspace and that the sidebar has changed in order to halt the call. These two conditions are very important to note at first glance.